openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -3526,31 +3529,34 @@ impl<T: CpuIo> UhHypercallHandler<'_, '_, T, TdxBacked> {
         Ok(())
     }
 
-    fn wake_processors_for_tlb_flush(&mut self, vtl: GuestVtl, processor_set: Option<Vec<u32>>) {
+    fn wake_processors_for_tlb_flush(
+        &mut self,
+        target_vtl: GuestVtl,
+        processor_set: Option<Vec<u32>>,
+    ) {
         match processor_set {
             Some(processors) => {
                 self.wake_processors_for_tlb_flush_inner(
-                    vtl,
+                    target_vtl,
                     processors.into_iter().map(|x| x as usize),
                 );
             }
-            None => self.wake_processors_for_tlb_flush_inner(vtl, 0..self.vp.partition.vps.len()),
+            None => {
+                self.wake_processors_for_tlb_flush_inner(target_vtl, 0..self.vp.partition.vps.len())
+            }
         }
     }
 
     fn wake_processors_for_tlb_flush_inner(
         &mut self,
-        vtl: GuestVtl,
+        target_vtl: GuestVtl,
         processors: impl Iterator<Item = usize>,
     ) {
         for target_vp in processors {
-            // Use SeqCst ordering to ensure that we are observing the most
-            // up-to-date value from other VPs. Otherwise we might not send a
-            // wake to a VP in a lower VTL, which could cause TLB lock holders
-            // to be stuck waiting until the target_vp happens to switch into
-            // VTL 2.
+            // Use SeqCst ordering here to ensure we are fully synchronized with
+            // the TLB address flush list we just added to.
             if self.vp.vp_index().index() as usize != target_vp
-                && self.vp.shared.active_vtl[target_vp].load(Ordering::SeqCst) == vtl as u8
+                && self.vp.shared.active_vtl[target_vp].load(Ordering::SeqCst) == target_vtl as u8
```

Comment:
How does this work if the target VP is in VTL0 and there's a VTL1 flush request? Don't we still need to wake the target, so that the source VP doesn't stall in VTL1?